src/pudl/validate.py

```diff
@@ -76,8 +76,10 @@ def check_date_freq(df1: pd.DataFrame, df2: pd.DataFrame, mult: int) -> None:
         raise ValueError("Missing report_date column in one or both input DataFrames")
 
     # Remove ytd values that mess up ratio assumptions
-    if "data_maturity" in df2:
-        df2 = df2[df2["data_maturity"] != "incremental_ytd"].copy()
+    if "data_maturity" in df1.columns:
+        df1 = df1.loc[df1["data_maturity"] != "incremental_ytd"]
+    if "data_maturity" in df2.columns:
+        df2 = df2.loc[df2["data_maturity"] != "incremental_ytd"]
```

Comment:
I made the SQL version check both dataframes since that seems more robust in the event that we're using a different dataframe to compare against at some point, and wanted the python version to be the same for debugging purposes.